Remove shimoga dependency from Dolphin project

Currently our CI server fails to build Dolphin. This is because maven fails to resolve dependencies to `shimoga` even we included the jar file in the project under `/lib`. This problem does not happen in the local machines.

Below is the error log from CI.

```
[INFO] Downloading: http://repository.apache.org/snapshots/com/microsoft/reef/shimoga/0.1-SNAPSHOT/maven-metadata.xml
[WARNING] Could not transfer metadata com.microsoft.reef:shimoga:0.1-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (http://repository.apache.org/snapshots): TransferFailedException
[WARNING] Failure to transfer com.microsoft.reef:shimoga:0.1-SNAPSHOT/maven-metadata.xml from http://repository.apache.org/snapshots was cached in the local repository, resolution will not be reattempted until the update interval of apache.snapshots has elapsed or updates are forced. Original error: Could not transfer metadata com.microsoft.reef:shimoga:0.1-SNAPSHOT/maven-metadata.xml from/to apache.snapshots (http://repository.apache.org/snapshots): TransferFailedException
[INFO] Downloading: http://repository.apache.org/snapshots/com/microsoft/reef/shimoga/0.1-SNAPSHOT/shimoga-0.1-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] CAY ................................................ SUCCESS [  2.141 s]
[INFO] Dolphin ............................................ FAILURE [  0.644 s]
[INFO] Elastic Memory ..................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.487 s
[INFO] Finished at: 2015-07-31T16:28:52+09:00
[INFO] Final Memory: 29M/607M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project dolphin: Could not resolve dependencies for project edu.snu.cay:dolphin:jar:0.1-SNAPSHOT: Failed to collect dependencies at com.microsoft.reef:shimoga:jar:0.1-SNAPSHOT: Failed to read artifact descriptor for com.microsoft.reef:shimoga:jar:0.1-SNAPSHOT: Could not transfer artifact com.microsoft.reef:shimoga:pom:0.1-SNAPSHOT from/to apache.snapshots (http://repository.apache.org/snapshots): TransferFailedException: ClientProtocolException: The server failed to respond with a valid HTTP response -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :dolphin
```

To solve this, we can choose one of following:
1) Debug the problem
2) Remove Shimoga dependency

@dafrista, @jsjason and I agreed to go with 2, because there is no point to keep shimoga once `REEF-126` is merged.

--- a/dolphin/pom.xml
+++ b/dolphin/pom.xml
@@ -33,7 +33,6 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <shimoga.version>0.1-SNAPSHOT</shimoga.version>
     <hadoop.version>2.4.0</hadoop.version>
     <mahout.version>0.9</mahout.version>
   </properties>

